crates/example-types/src/block_types.rs

```diff
@@ -150,8 +146,9 @@ impl BlockPayload for TestBlockPayload {
         (Self::genesis(), ())
     }
 
-    fn encode(&self) -> Result<Self::Encode<'_>, Self::Error> {
-        Ok(TestTransaction::encode(self.transactions.clone())?.into_iter())
+    fn encode(&self) -> Result<Arc<[u8]>, Self::Error> {
+        let encoded = TestTransaction::encode(&self.transactions);
+        encoded.map(Arc::from)
```

Comment:
```suggestion
        TestTransaction::encode(&self.transactions).map(Arc::from)
```
code golf improvement? 😉